front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/handleParseFiles.ts

```diff
@@ -9,71 +9,94 @@ export const handleFileReadingError = (error: Error) => {
   console.error('File reading error:', error);
 };
 
-export const handleJsonParsingError = (error: Error, dispatch: Dispatch) => {
-  console.error('Error parsing JSON:', error);
-  dispatch(
-    setFailure({
-      name: t('errorMessages.error'),
-      message: t('errorMessages.errorInvalidJSONFormat'),
-    })
-  );
-};
+const validateTrainSchedules = (
+  importedTrainSchedules: Record<string, unknown>[]
+): TrainScheduleBase[] | null => {
+  const isInvalidTrainSchedules = importedTrainSchedules.some((trainSchedule) => {
+    if (
+      ['constraint_distribution', 'path', 'rolling_stock_name', 'start_time', 'train_name'].some(
+        (key) => !(key in trainSchedule)
+      ) ||
+      !Array.isArray(trainSchedule.path)
+    ) {
+      return true;
+    }
```

Comment:
Might be safer (in case of misspell) to do that
```suggestion
    const compulsoryKeys: (keyof TrainScheduleBase)[] = [
      'constraint_distribution',
      'path',
      'rolling_stock_name',
      'start_time',
      'train_name',
    ];
    if (
      compulsoryKeys.some((key) => !(key in trainSchedule)) ||
      !Array.isArray(trainSchedule.path)
    ) {
      return true;
    }
```

Reply:
Done :+1: